### PR TITLE
Rework BMW docs for new quality scale

### PR DESCRIPTION
**Pull request by rikroe**
## Proposed change

Rework BMW docs for quality scale.

This contains the changes from #36032, so I wasn't sure if it should be based against `next` or `current`, but as it will get merged into `current` with 2024.12, I hope this is correct.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Updated documentation to reflect the rebranding from BMW Connected Drive to MyBMW.
	- Added prerequisites for MyBMW account and MINI vehicle registration.
	- Enhanced configuration instructions for username, password, region, and captcha token.
	- Introduced a new section for custom polling intervals and detailed platform functionalities.
	- Expanded troubleshooting section with common issues and recovery steps.
	- Documented known limitations regarding entity availability and charging control functionalities.

### CODE REVIEW

File: source/_integrations/bmw_connected_drive.markdown [hunk 1]
```diff
@@ -56,23 +121,17 @@ This integration provides the following platforms:
...
-|---------|-------------|
-| Read-only | No execution of actions to the vehicle. Still possible to send messages and POIs via `notify` and to request a status update via `bmw_connected_drive.update_state`.
+**Use with care!**
+{% endwarning %}
 
-## Notifications
+### Notifications
 
-The `bmw_connected_drive` integration offers a notification action. Using this action you can send Points of Interest (POI) to your vehicle. In your vehicle you can select this POI and the navigation will automatically start using the POI as a destination.
+The **BMW Connected Drive** integration offers a notification action. Using this action you can send Points of Interest (POI) to your vehicle. In your vehicle you can select this POI and the navigation will automatically start using the POI as a destination.
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
The **BMW Connected Drive** integration offers a notification action. Using this action you can send Points of Interest (POI) to your vehicle. In your vehicle, you can select this POI, and the navigation will automatically start using the POI as a destination.
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 2]
```diff
@@ -31,20 +31,85 @@ ha_platforms:
...
-The **BMW Connected Drive** {% term integration %} also works with (recent) Mini vehicles. You need to have a working Mini Connected account, and a Mini Connected enabled vehicle for this to work.
+The **BMW Connected Drive** {% term integration %} lets you retrieve data of your BMW or MINI vehicle from the MyBMW portal (previously BMW Connected Drive).
 
 {% note %}
 The {% term entities %} available in Home Assistant heavily depend on your vehicle's capabilities (model year, headunit, etc.). The integration will make sure all available car attributes are added as entities.
 {% endnote %}
 
+## Prerequisites
+
+You need to have an active MyBMW account with a connected car. For MINI vehicles, you register your with MINI Connected.
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
You need to have an active MyBMW account with a connected car. For MINI vehicles, you register with MINI Connected.
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 3]
```diff
@@ -31,20 +31,85 @@ ha_platforms:
...
+  required: true
+  type: string
+Region:
+  description: "Region of your MyBMW/MINI Connected account."
+  required: true
+  type: select
+  options: china, north_america, rest_of_world
+Captcha token (second step, only for North America and Rest of World):
+  description: |
+    The **North America** and **Rest of World** regions require a captcha challenge to be solved, i.e. you need to verify that you are not a robot.
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
    The **North America** and **Rest of World** regions require a captcha challenge to be solved. For example, you need to verify that you are not a robot.
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 4]
```diff
@@ -31,20 +31,85 @@ ha_platforms:
...
+|---------------|------------|
+| China         | 5 minutes  |
+| North America | 10 minutes |
+| Rest of world | 5 minutes  |
+
+{% note %}
+This will only refresh data from the BMW/MINI servers and **not** from your car.
+
+- Combustion engine cars typically only update data from car to server when the car is parked and the engine is shut off.
+- Electric vehicles typically only update while the vehicle is parked and turned off or while charging
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
- Electric vehicles typically only update while the vehicle is parked and turned off or while they are charging.
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]
Tiny tweaks

**Comment by c0ffeeca7** [comment]
To your question: It belongs on the `next` branch. PRs with a parent PR go on next, so that they can be merged together in the same release. If this docs PR were merged no to current, the docs would not be correct right now.

**Comment by rikroe** [comment]
> To your question: It belongs on the `next` branch. PRs with a parent PR go on next, so that they can be merged together in the same release. If this docs PR were merged no to current, the docs would not be correct right now.

Thanks!

I'd probably still keep it on `current`, as parent + docs PR for `next` is already merged and will be live on wednesday. I'd rather rebase this afterwards.

**Comment by rikroe** [comment]
With 2024.12, the target of this PR is now current as all relevant core changes have been released.

File: source/_integrations/bmw_connected_drive.markdown [hunk 5]
```diff
@@ -138,32 +192,53 @@ If you do not want this, trigger the `vehicle_finder` action from your phone and
...
 
-Using these selects will impact the state of your vehicle, use them with care!
-
 - **Target SoC**: Vehicle will charge until this battery level is reached. Not available on all EVs.
 
+## Troubleshooting
+{% details "Problem: Invalid authentication" %}
+
+This can happen during initial login or after some time. Please do the following steps:
+- Log in to your MyBMW **website** and verify your credentials (i.e. ensure that username and password are correct).
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
- Log in to your MyBMW **website** and verify your credentials (for example, ensure that username and password are correct).
```
following MS Style Guide, we [avoid e.g.](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg)

File: source/_integrations/bmw_connected_drive.markdown [hunk 6]
```diff
@@ -138,32 +192,53 @@ If you do not want this, trigger the `vehicle_finder` action from your phone and
...
+{% details "Problem: Captcha validation missing" %}
+
+Sometimes, your account can be force-logged-out. For **North America** and **Rest of World**, the recovery requires manual intervention.
+
+Home Assistant will show a repair issue to **reconfigure** the integration. Follow the steps to log in again.
+
+{% enddetails %}
+
+## Known limitations
+- The entities available to Home Assistant depend on your vehicle. Even inside the same model code (e.g. U11 for BMW X1) you will see different entities, depending on your specific car's features.
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
- The entities available to Home Assistant depend on your vehicle. Even inside the same model code (for example, U11 for BMW X1) you will see different entities, depending on your specific car's features.
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 7]
```diff
@@ -138,32 +192,53 @@ If you do not want this, trigger the `vehicle_finder` action from your phone and
...
+- Not all features, mostly related to charging control for (PH)EVs, are implemented. If you have a functionality in your MyBMW/MINI app that is not yet available, search for an existing feature request in the [`bimmer_connected` discussions](https://github.com/bimmerconnected/bimmer_connected/discussions) or create a new one.
+
+## Remove integration
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
## Removing the integration
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 8]
```diff
@@ -138,32 +192,53 @@ If you do not want this, trigger the `vehicle_finder` action from your phone and
...
+
+{% details "Problem: Captcha validation missing" %}
+
+Sometimes, your account can be force-logged-out. For **North America** and **Rest of World**, the recovery requires manual intervention.
+
+Home Assistant will show a repair issue to **reconfigure** the integration. Follow the steps to log in again.
+
+{% enddetails %}
+
+## Known limitations
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown:227]:
```suggestion
## Known limitations

```

File: source/_integrations/bmw_connected_drive.markdown [hunk 9]
```diff
@@ -31,20 +31,87 @@ ha_platforms:
...
+{% configuration_basic %}
+Username:
+  description: |
+    Username of your MyBMW/MINI Connected account.
+
+    &nbsp;
+    
+    **China**: Your username/phone number must be prefixed with `86`, i.e. `8612345678`.
+  required: true
+  type: string
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
```
removing these, as `required` and `type` are not displayed with the configuration_basic tag

File: source/_integrations/bmw_connected_drive.markdown [hunk 10]
```diff
@@ -31,20 +31,87 @@ ha_platforms:
...
+Password:
+  description: "Password of your MyBMW/MINI Connected account."
+  required: true
+  type: string
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 11]
```diff
@@ -31,20 +31,87 @@ ha_platforms:
...
+Region:
+  description: "Region of your MyBMW/MINI Connected account."
+  required: true
+  type: select
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 12]
```diff
@@ -31,20 +31,87 @@ ha_platforms:
...
+  description: |
+    The **North America** and **Rest of World** regions require a captcha challenge to be solved, that means you need to verify that you are not a robot.
+
+    - After entering your login data, a second step will ask for a `Captcha token` and provide you with a **link** to a website. 
+    - Open this link, solve the **"Are you a human?"** challenge and press **Submit**.
+    - Copy the resulting token into Home Assistant and continue.
+
+    No data of your Home Assistant instance is shared with any third party during this step.
+  required: true
+  type: string
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 13]
```diff
@@ -138,32 +184,54 @@ If you do not want this, trigger the `vehicle_finder` action from your phone and
...
-## Numbers
+### Numbers
 
 If you have a (PH)EV, you can control the charging process through Home Assistant. The number entities are created automatically depending on your vehicle's capabilities and can be changed from the UI or using the `number.set_value` action. For more information, please see the [number documentation](/integrations/number/).
 
-Using these selects will impact the state of your vehicle, use them with care!
-
 - **Target SoC**: Vehicle will charge until this battery level is reached. Not available on all EVs.
 
+## Troubleshooting
```
Comment (c0ffeeca7) [inline source/_integrations/bmw_connected_drive.markdown:207]:
```suggestion
## Troubleshooting

```

**Review by c0ffeeca7 — APPROVED** [review]
Nice addition to the docs! Thank you, @rikroe :+1:

**Pull request merged**